File: array/Merge_Sorted_Array.c
#include <stdio.h>
#include <stdlib.h>

int* merge_sorted_arrays(int* a, int n, int* b,int m, int* out_size){

	int i=0,j=0,k=0;

	// Allocate the result array
	int* merged = malloc((n+m) *sizeof(int));
	
	if (!merged) return NULL;

	// Merge while both arrays have the elements
	while (i<n && j<m){
		if (a[i]<b[j]){
			merged[k++] = a[i++];
		}
		else if (a[i] > b[j]){
			merged[k++]= b[j++];
		}else{
			// Equivalent of python merged_array.extend()
			merged[k++] = a[i++];
			merged[k++] = b[j++];
		}
	}
// Equivalent of merged_array.extend(a[i:])
while (i<n){
	merged[k++] = a[i++];
}

// Equivalent of merged_array.extend(b[j:])
while (j<m){
	merged[k++]= b[j++];
}

*out_size = k;
return merged;

}

int main() {
    int a[] = {1, 3, 5};
    int b[] = {2, 3, 4};

    int out_size;
    int* result = merge_sorted_arrays(a, 3, b, 3, &out_size);

    for (int i = 0; i < out_size; i++) {
        printf("%d ", result[i]);
    }

    free(result);
    return 0;
}

File: array/Merge_Sorted_Array.py
def merge_sorted_array(a,b):
	merged_array,i,j = [],0,0

	while i<len(a) and j<len(b):
		if a[i]<b[j]:
			merged_array.append(a[i])
			i += 1
		elif a[i]>b[j]:
			merged_array.append(b[j])
			j +=1
		else:
			merged_array.extend([a[i],b[j]])
			i,j = i+1, j+1

	merged_array.extend(a[i:])
	merged_array.extend(b[i:])

	return merged_array
array1 = [1, 3, 5, 7, 9]
array2 = [2, 4, 6, 8, 10]
print(merge_sorted_array(array1, array2))
